### PR TITLE
For #16162: Simplify site addresses in saved logins view

### LINKED ISSUE #16162: Shorten the name of the protocol (https) in the site address in the list of Logins
The name of the protol https:// is not needed in the Logins list because:
1. Occupies space (especially on narrow screens)
2. Passwords are secret information that should be transmitted by https by default, so you should specify the protocol when it is different from https.

Screenshot_2020-10-23-13-39-53

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-8488)

### PR DESCRIPTION
**Pull request by fice-t**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #16162

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt [hunk 1]
```diff
@@ -51,6 +51,17 @@ class LoginsListViewHolderTest {
...
+        // test for proper simplification of mobile URLs
+        val mLogin = baseLogin.copy(origin = "https://m.mozilla.org")
+        holder.bind(mLogin)
+
+        assertEquals("mozilla.org", binding.webAddressView.text)
+
+        val mobileLogin = baseLogin.copy(origin = "https://mobile.mozilla.org")
+        holder.bind(mobileLogin)
+
+        assertEquals("mozilla.org", binding.webAddressView.text)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt:64]: Could you add this as a separate @Test?

**Comment by gabrielluong** [comment]
Could you also add a before and after screenshot of your changes? Thanks!

File: app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt (hunk 1 above)
Comment (fice-t) [reply app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt]: Done. I also factored out the holder definition.

**Comment by fice-t** [comment]
Before and after:

**Comment by gabrielluong** [comment]
> Before and after:
> 
> 

@topotropic Any chance you could take a quick look and provide any necessary UX feedback?

File: app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt [hunk 2]
```diff
@@ -25,40 +25,48 @@ class LoginsListViewHolderTest {
...
-            interactor,
-        )
         holder.bind(baseLogin)
 
         assertEquals("mozilla.org", binding.webAddressView.text)
         assertEquals("admin", binding.usernameView.text)
     }
 
+    @Test
+    fun `GIVEN url to bind WHEN url has a mobile prefix THEN mobile prefix is stripped`() {
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt:58]:
```suggestion
    fun `GIVEN url has a mobile prefix WHEN url is binded THEN mobile prefix is stripped`() {
```

**Review by gabrielluong — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt (hunk 2 above)
Comment (fice-t) [reply app/src/test/java/org/mozilla/fenix/settings/logins/LoginsListViewHolderTest.kt]: Updated.

**Comment by gabrielluong** [comment]
@fice-t We probably won't land this until Monday.

**Review by topotropic — APPROVED** [review]

**Comment by jonalmeida** [comment]
Based on the feedback from Mergify, we need to close this PR in order to merge it in the PR that has the full CI build running on it.

Thank you for changes and you can watch it land in https://github.com/mozilla-mobile/fenix/pull/27530. 🙂

**Pull request closed without merging**